src/Analyser/TypeSpecifier.php

```diff
@@ -796,7 +797,11 @@ public function specifyTypesInCondition(
 					}
 				}
 
-				$asserts = $methodReflection->getAsserts()->mapTypes(static fn (Type $type) => TemplateTypeHelper::resolveTemplateTypes($type, $parametersAcceptor->getResolvedTemplateTypeMap()));
+				$templateTypeMap = new TemplateTypeMap(array_merge(
+					$methodReflection->getDeclaringClass()->getActiveTemplateTypeMap()->getTypes(),
+					$parametersAcceptor->getResolvedTemplateTypeMap()->getTypes(),
```

Comment:
For some reason `$parametersAcceptor->getResolvedTemplateTypeMap()` does not contain the class template types when the method also has a template type. This might be a symptom of an underlying issue, but this does resolve my issue :')